bokehjs/src/lib/models/widgets/file_input.ts

```diff
@@ -36,22 +36,38 @@ export class FileInputView extends InputWidgetView {
     })
   }
 
+  override connect_signals(): void {
+    super.connect_signals()
+    const {_clear_input} = this.model.properties
+    this.on_change(_clear_input, this.clear_input)
+  }
+
   async load_files(files: FileList): Promise<void> {
     const values: string[] = []
     const filenames: string[] = []
     const mime_types: string[] = []
+    const {directory, multiple, accept} = this.model
 
     for (const file of files) {
       const data_url = await this._read_file(file)
       const [, mime_type="",, value=""] = data_url.split(/[:;,]/, 4)
 
-      values.push(value)
-      filenames.push(file.name)
-      mime_types.push(mime_type)
+      if (directory) {
+        const ext = file.name.split(".").pop()
+        if ((!is_nullish(accept) && isString(ext)) ? accept.includes(`.${ext}`) : true) {
```

Comment:
This check will likely need some more thought. 

What comes to mind is that it should also check the `mime_type`.